src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs

```diff
@@ -335,6 +339,12 @@ public IActionResult BuildAndSendCreateSmartContractTransaction([FromBody] Build
 
             Transaction transaction = this.network.CreateTransaction(response.Hex);
 
+            if (!this.connectionManager.ConnectedPeers.Any())
+            {
+                this.logger.LogTrace("(-)[NO_CONNECTED_PEERS]");
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.Forbidden, "Can't send transaction: sending transaction requires at least one connection!", string.Empty);
```

Comment:
Not sure if the `HttpStatusCode` should be forbidden? I mean, we did infact make a valid API call, its just that an internal component wasn't able to provide the requested data? Http status codes is relevant to the call and not the contents of the it.

`HTTP Status Code 403: The server understood the request but refuses to authorize it.`

Reply:
I would return an OK result with the message in it as the call "succeeded".

Reply:
I just borrowed the implementation from `WalletController`

Reply:
I don't disagree, but the normal wallet controller currently returns forbidden.

Reply:
OK lets fix this in another PR.